unittests/kv_tests.cpp

```diff
@@ -701,90 +688,59 @@ class kv_tester : public tester {
       BOOST_REQUIRE_EQUAL("", push_action(N(itlifetime), mvo()("db", N(eosio.kvdisk))));
-      test_basic(N(eosio.kvram));
-      */
       test_basic(N(eosio.kvdisk));
    }
 
    void test_kv_scan_common() { //
       // four possibilities depending on whether the next or previous contract table has elements
-      test_scan(N(eosio.kvram), N(kvtest2));
-      test_scan(N(eosio.kvram), N(kvtest4));
-      test_scan(N(eosio.kvram), N(kvtest3));
-      test_scan(N(eosio.kvram), N(kvtest1));
       test_scan(N(eosio.kvdisk), N(kvtest2));
       test_scan(N(eosio.kvdisk), N(kvtest4));
       test_scan(N(eosio.kvdisk), N(kvtest3));
       test_scan(N(eosio.kvdisk), N(kvtest1));
    }
 
-   void test_kv_scan_ram_after_disk_common() {
-      // Make sure that the behavior of one database is not affected by having the other database populated.
-      test_scan(N(eosio.kvdisk), N(kvtest2));
-      test_scan(N(eosio.kvdisk), N(kvtest4));
-      test_scan(N(eosio.kvdisk), N(kvtest3));
-      test_scan(N(eosio.kvdisk), N(kvtest1));
-      test_scan(N(eosio.kvram), N(kvtest2));
-      test_scan(N(eosio.kvram), N(kvtest4));
-      test_scan(N(eosio.kvram), N(kvtest3));
-      test_scan(N(eosio.kvram), N(kvtest1));
-   }
-   
    void test_kv_scanrev_common() {
-      test_scanrev(N(eosio.kvram), N(kvtest2));
-      test_scanrev(N(eosio.kvram), N(kvtest4));
-      test_scanrev(N(eosio.kvram), N(kvtest3));
-      test_scanrev(N(eosio.kvram), N(kvtest1));
       test_scanrev(N(eosio.kvdisk), N(kvtest2));
       test_scanrev(N(eosio.kvdisk), N(kvtest4));
       test_scanrev(N(eosio.kvdisk), N(kvtest3));
       test_scanrev(N(eosio.kvdisk), N(kvtest1));
    }
    
-   void test_kv_scanrev_ram_after_disk_common() {
-      test_scanrev(N(eosio.kvdisk), N(kvtest2));
-      test_scanrev(N(eosio.kvdisk), N(kvtest4));
-      test_scanrev(N(eosio.kvdisk), N(kvtest3));
-      test_scanrev(N(eosio.kvdisk), N(kvtest1));
-      test_scanrev(N(eosio.kvram), N(kvtest2));
-      test_scanrev(N(eosio.kvram), N(kvtest4));
-      test_scanrev(N(eosio.kvram), N(kvtest3));
-      test_scanrev(N(eosio.kvram), N(kvtest1));
-   }
-   
    void test_kv_scanrev2_common() {
-      test_scanrev2(N(eosio.kvram));
       test_scanrev2(N(eosio.kvdisk));
    }
    
    void test_kv_iterase_common() {
-      test_iterase(N(eosio.kvram));
       test_iterase(N(eosio.kvdisk));
    }
    
    void test_kv_ram_usage_common() {
-      test_ram_usage(N(eosio.kvram));
       test_ram_usage(N(eosio.kvdisk));
    }
    
    void test_kv_resource_limit_common() { //
-      test_resource_limit(N(eosio.kvram));
       test_resource_limit(N(eosio.kvdisk));
    }
    
    void test_kv_key_value_limit_common() { //
-      test_key_value_limit(N(eosio.kvram));
       test_key_value_limit(N(eosio.kvdisk));
    }
 
    abi_serializer abi_ser;
    abi_serializer sys_abi_ser;
 };
 
+class kv_rocksdb_tester : public kv_tester {
```

Comment:
get rid of this class and add get_config to kv_tester constructor and pass in backing_store, then you can use BOOST_DATA_TEST_CASE_F to pass in CHAINBASE and ROCKSDB as the dataset (instead of databases, which isn't needed anymore).

Reply:
also, we won't need all these helper methods anymore, so move the BOOST_.. macros instead of the functions that surround the implemenation.

Reply:
Thanks for letting me know about those techniques. I will utilize them in future tests.